rs/execution_environment/src/execution_environment/tests/canister_snapshots.rs

```diff
@@ -1714,3 +1714,91 @@ fn load_canister_snapshot_charges_canister_cycles() {
         test.canister_state(canister_id).system_state.balance() < initial_balance - expected_charge
     );
 }
+
+#[test]
+fn snapshot_must_include_globals() {
+    let wat = r#"
+    (module
+        (import "ic0" "msg_reply" (func $msg_reply))
+        (import "ic0" "msg_reply_data_append"
+          (func $msg_reply_data_append (param i32 i32)))
+      
+        (func $read_global
+          (i32.store
+            (i32.const 0)
+            (global.get 0)
+          )
+          (call $msg_reply_data_append
+            (i32.const 0)
+            (i32.const 4))
+          (call $msg_reply)
+        )
+      
+        (func $increase_global
+          (global.set 0
+            (i32.add
+              (global.get 0)
+              (i32.const 1)
+            )
+          )
+          (call $msg_reply)
+        )
+      
+        (memory $memory 1)
+        (export "memory" (memory $memory))
+        (global (export "counter") (mut i32) (i32.const 0))
+        (export "canister_query read_global" (func $read_global))
+        (export "canister_update increase_global" (func $increase_global))
+      )"#;
+    let wasm = wat::parse_str(wat).unwrap();
+
+    const CYCLES: Cycles = Cycles::new(1_000_000_000_000_000);
+    let own_subnet = subnet_test_id(1);
+    let caller_canister = canister_test_id(1);
```

Comment:
nit: I'm not sure if the specific subnet and caller, as well as the reserved cycles, make a difference for this test. But we can clean them up later...

Reply:
Thank you! I simplified the test.